client/src/components/Profile/ProfileDesktop.js

```diff
@@ -0,0 +1,133 @@
+import React, {
+  useEffect,
+  useReducer,
+  useState,
+  useCallback,
+  useRef,
+} from "react";
+import { useTranslation } from "react-i18next";
+
+import styled from "styled-components";
+import { Menu } from "antd";
+import { WhiteSpace } from "antd-mobile";
+
+import GTM from "constants/gtm-tags";
+import { SeeAllWrapper } from "components/Feed/FeedWrappers";
+import { ChildMenuWrapper } from "components/Profile/ProfileComponents";
+import { theme, mq } from "constants/theme";
+import Activity from "components/Profile/Activity";
+
+const TAB_TYPE = {
+  POSTS: {
+    REQUESTS: "Requests",
+    OFFERS: "Offers",
+  },
+  REQUESTS: {
+    ACTIVE_REQS: "Active",
+    ARCHIVED_REQS: "Archived",
+    DRAFTS_REQS: "Drafts",
+  },
+  OFFERS: {
+    ACTIVE_OFRS: "Active",
+    ARCHIVED_OFRS: "Archived",
+    DRAFTS_OFRS: "Drafts",
+  },
+};
+const gtmTagsMap = {
+  REQUESTS: `${GTM.profile.requests}`,
+  OFFERS: `${GTM.profile.offers}`,
+};
+
+const gtmTag = (tag) => GTM.profile.viewProfilePrefix + tag;
+
+const ProfileDesktop = ({
+  isAuthenticated,
+  isOrg,
+  // profileId: pathUserId,
+  setInternalTab,
+  user,
+  menuView: viewType,
+  profilePost,
+  postDispatch,
+  postDelete,
+  handlePostDelete,
+  deleteModalVisibility,
+  handleCancelPostDelete,
+  loadNextPage,
+  isLoading,
+  isItemLoaded,
+  itemCount,
+  loadMore,
+  totalPostCount,
+  isMobile,
+}) => {
+  const { t } = useTranslation();
+
+  const defaultState = useCallback(() => {
+    if (viewType === "POSTS") return "REQUESTS";
+    else if (viewType === "REQUESTS") return "ACTIVE_REQS";
+    else return "ACTIVE_OFRS";
+  });
+  const [childTab, setChildTab] = useState(defaultState);
+
+  const handleTabChange = (e) => {
+    // console.log("tab Change", e.key);
+    setChildTab(e.key);
+    setInternalTab(e.key);
+  };
+
+  useEffect(() => {
```

Comment:
why do we need to useEffect?

Reply:
when we change parent Tabs, we want to set first tab (child) as default active tab. and it should trigger fetch too.